commands/conduit_cli.py

```diff
@@ -38,144 +55,117 @@ def get_cluster_arn(app: str, env: str) -> str:
+    connection_secret_id = f"/copilot/{app}/{env}/secrets/{name}"
 
-def update_postgres_command(app: str, env: str, command: str, secret_name: str) -> str:
-    secret = get_postgres_secret(app, env, secret_name)
-    secret_arn = secret["ARN"]
-    secret_json = json.loads(secret["SecretString"])
-    postgres_password = secret_json["password"]
+    secrets_manager = boto3.client("secretsmanager")
+    ssm = boto3.client("ssm")
 
-    return f"{command} --secrets DB_SECRET={secret_arn} --env-vars POSTGRES_PASSWORD={postgres_password}"
+    try:
+        return secrets_manager.describe_secret(SecretId=connection_secret_id)["ARN"]
+    except secrets_manager.exceptions.ResourceNotFoundException:
+        pass
 
+    try:
+        return ssm.get_parameter(Name=connection_secret_id, WithDecryption=False)["Parameter"]["ARN"]
+    except ssm.exceptions.ParameterNotFound:
+        pass
 
-def create_task(app: str, env: str, addon_type: str, secret_name: str) -> None:
-    command = f"copilot task run -n tunnel-{addon_type} --image {CONDUIT_IMAGE_LOCATIONS[addon_type]} --app {app} --env {env}"
-    if addon_type == "postgres":
-        command = update_postgres_command(app, env, command, secret_name)
+    raise SecretNotFoundConduitError(name)
 
-    subprocess.call(command, shell=True)
 
+def create_addon_client_task(app: str, env: str, addon_type: str, addon_name: str):
+    connection_secret_arn = get_connection_secret_arn(app, env, addon_name.upper())
 
-def is_task_running(cluster_arn: str, addon_type: str) -> bool:
-    tasks = boto3.client("ecs").list_tasks(
-        cluster=cluster_arn, desiredStatus="RUNNING", family=f"copilot-tunnel-{addon_type}"
+    subprocess.call(
+        f"copilot task run --app {app} --env {env} --task-group-name conduit-{addon_name} "
+        f"--image {CONDUIT_DOCKER_IMAGE_LOCATION}:{addon_type} "
+        f"--secrets CONNECTION_SECRET={connection_secret_arn} "
+        "--platform-os linux "
+        "--platform-arch arm64",
+        shell=True,
     )
 
-    try:
-        if tasks["taskArns"]:
-            described_tasks = boto3.client("ecs").describe_tasks(cluster=cluster_arn, tasks=tasks["taskArns"])
-            agent_running = False
-
-            # The ExecuteCommandAgent often takes longer to start running than the task and without the agent it's not possible to exec into a task.
-            for agent in described_tasks["tasks"][0]["containers"][0]["managedAgents"]:
-                if agent["name"] == "ExecuteCommandAgent" and agent["lastStatus"] == "RUNNING":
-                    agent_running = True
 
-            return described_tasks["tasks"][0]["lastStatus"] == "RUNNING" and agent_running
+def addon_client_is_running(cluster_arn: str, addon_name: str) -> bool:
+    tasks = boto3.client("ecs").list_tasks(
+        cluster=cluster_arn,
+        desiredStatus="RUNNING",
+        family=f"copilot-conduit-{addon_name}",
+    )
 
-    except ValueError:
+    if not tasks["taskArns"]:
```

Comment:
Minor one, you could probably just do

```
return not tasks["taskArns"]
```

Reply:
This is a guard clause to stop execution if task arns is empty, for the next line where we call describe tasks. We could refactor to a try except return false if that makes it an easier read.